buffer/src/main/java/io/netty/buffer/PooledSlicedByteBuf.java

```diff
@@ -114,17 +114,14 @@ public ByteBuf retainedSlice(int index, int length) {
     @Override
     public ByteBuf duplicate() {
         // Capacity is not allowed to change for a sliced ByteBuf, so length == capacity()
-        final ByteBuf duplicate = unwrap().slice(adjustment, capacity());
+        final ByteBuf duplicate = super.slice(adjustment, capacity());
```

Comment:
What would you expect `c` to be in the following situation:

```java
ByteBuf a = {1, 2, 3, 4, 5, 6, 7, 8}; // pseudo code
ByteBuf b = a.retainedSlice(a.readerIndex() + 1, 6); // {2, 3, 4, 5, 6, 7}
ByteBuf c = b.duplicate();

// c should represent {2, 3, 4, 5, 6, 7} because it is a duplicate of b right?
```